R-packages/covidcast/tests/testthat/test-covidcast.R

```diff
@@ -1,25 +1,42 @@
 library(covidcast)
+library(httptest)
 library(mockery)
 
-test_that("covidcast_meta formats result correctly", {
-  stub(covidcast_meta, ".request",
-       list(message = "success",
-            epidata = data.frame(
-              min_time = c("20200101", "20201002"),
-              max_time = c("20200102", "20201003"),
-              max_issue = c("20200404", "20201101"))))
-
-  # ensure columns that should be converted to dates are indeed dates
-  expect_equal(covidcast_meta(),
-               structure(
-                 data.frame(
-                   min_time = as.Date(c("2020-01-01", "2020-10-02")),
-                   max_time = as.Date(c("2020-01-02", "2020-10-03")),
-                   max_issue = as.Date(c("2020-04-04", "2020-11-01"))
-                 ),
-                 class = c("covidcast_meta", "data.frame")
-               ))
+# Many of these tests use mockery::with_mock_api. This replaces calls to the
+# live API server, instead returning static JSON files from disk. Three
+# weaknesses to be aware of:
+#
+# 1. The HTTP URL and parameters are hashed into a single filename to use. This
+# means that you can't have the query return one thing, then have the same query
+# return something else, for instance to test how the client handles different
+# types of errors.
+#
+# 2. Once you've written a test, it can be difficult to find the file storing
```

Comment:
Can you also add comments in the json files to indicate to which query they belong?

Reply:
what's your preferred way for adding comments in JSON files? a dummy key/value pair?

Reply:
Ah didn't realize json didn't support comments.  Ignore it then.